connector/saml/saml_test.go

```diff
@@ -22,21 +28,82 @@ func loadCert(ca string) (*x509.Certificate, error) {
 	return x509.ParseCertificate(block.Bytes)
 }
 
-func TestVerify(t *testing.T) {
-	cert, err := loadCert("testdata/okta-ca.pem")
+func runVerify(t *testing.T, ca string, resp string, shouldSucceed bool) {
+	cert, err := loadCert(ca)
 	if err != nil {
 		t.Fatal(err)
 	}
 	s := certStore{[]*x509.Certificate{cert}}
 
-	validator := sdig.NewDefaultValidationContext(s)
+	validator := dsig.NewDefaultValidationContext(s)
 
-	data, err := ioutil.ReadFile("testdata/okta-resp.xml")
+	data, err := ioutil.ReadFile(resp)
 	if err != nil {
 		t.Fatal(err)
 	}
 
 	if _, err := verify(validator, data); err != nil {
+		if shouldSucceed {
+			t.Fatal(err)
+		}
+	} else {
+		if !shouldSucceed {
+			t.Fatalf("expected an invalid signatrue but verification has been successful")
+		}
+	}
+}
+
+func TestVerify(t *testing.T) {
+	runVerify(t, "testdata/okta-ca.pem", "testdata/okta-resp.xml", true)
+}
+
+func TestVerifySignedMessageAndUnsignedAssertion(t *testing.T) {
+	runVerify(t, "testdata/idp-cert.pem", "testdata/idp-resp-signed-message.xml", true)
+}
+
+func TestVerifyUnsignedMessageAndSignedAssertion(t *testing.T) {
+	runVerify(t, "testdata/idp-cert.pem", "testdata/idp-resp-signed-assertion.xml", true)
+}
+
+func TestVerifySignedMessageAndSignedAssertion(t *testing.T) {
+	runVerify(t, "testdata/idp-cert.pem", "testdata/idp-resp-signed-message-and-assertion.xml", true)
+}
+
+func TestVerifyUnsignedMessageAndUnsignedAssertion(t *testing.T) {
+	runVerify(t, "testdata/idp-cert.pem", "testdata/idp-resp.xml", false)
+}
+
+func TestHandlePOST(t *testing.T) {
+	logger := logrus.New()
+	p := &provider{
+		issuer:       "http://localhost:5556/dex/callback",
+		ssoURL:       "http://idp.org/saml/sso",
+		now:          time.Now,
```

Comment:
We generally don't use `time.Now` in tests. At most

```
now := time.Now()

timeFunc := func() time.Time { return now }
```

That way we can ensure the test doesn't depend on some arbitrary amount of time having passed (or not passed) between components of the test.